makePoplogCommander.sh

```diff
@@ -476,46 +664,76 @@ void extendPath( char * prefix, char * path, char * suffix ) {
     free( buff );
 }
 
-void setUpEnvironment( char * base, int flags, Chain envv ) {
+****
+
+# Here we create three functions for setting the environment variables that
+# are unique to the build variants: nox, xm, xt. These will be called
+# nox_setUpEnvVars, xm_setUpEnvVars, xt_setUpEnvVars.
+VARIANTS='nox xt xm'
+for variant in $VARIANTS
+do
+    echo "void ${variant}_setUpEnvVars( char * base, bool inherit_env ) {"
+    cat _build/environments/$variant.env \
+    | sed -e 's/"/\\"/g' \
+    | sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'
+    echo "}"
+    echo
+done
+
+cat << \****
+// This function will establish the environment variables for Poplog.
+void setUpEnvironment( char * base, int flags, Vector envv ) {
     bool inherit_env = ( flags & INHERIT_ENV ) != 0;
     bool run_init_p = ( flags & RUN_INIT_P ) != 0;
 
     setenv( "usepop", base, !inherit_env );
+
+    int vflags = flags & VARIANT_FLAGS;
+    switch ( vflags ) {
 ****
-echo
 
-################################################################################
-# We now run the popenv.sh script inside a clean environment to capture the 
-# set of environment variables needed. We then need to replace any matches
-# of the string "_build/poplog_base" ($usepop) with our unique value USEPOP.
-# This will allow us to dynamically substitute with the selfHome'd value
-# at run-time. 
-#
-# This is inherently a weak strategy because it relies on being able to 
-# identify substitutions of $usepop. We improve its robustness by doing the
-# process twice with different values of $usepop - using the '..' trick.
-# If the resultant code is not identical we have a problem and we halt.
-# (N.B It is probably not necessary to run the variables through sort but I 
-# couldn't find a clear guarantee that env generates a sorted list.)
-################################################################################
+for variant in $VARIANTS
+do
+    echo "        case VARIANT_${variant^^}:"
+    echo "            ${variant}_setUpEnvVars( base, inherit_env );"
+    echo "            break;"
+done
 
-CODE1=`env -i sh -c '(usepop="_build/poplog_base" && . $usepop/pop/com/popenv.sh && env)' | sort \
-| grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' \
-| sed -e 's!_build/poplog_base![//USEPOP//]!g' \
-| sed -e 's/"/\\"/g' \
-| sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'`
+# We have to default to 'xt' at the moment because the 'xm' build has issues.
```

Comment:
No need for this comment. The code is clear and the comment is wrong (since we use `xm`)